app/main.py

```diff
@@ -1,3 +1,18 @@
 def cache(func):
-    # Write your code here
-    pass
+    cached_values_list = []
+
+    def inner(*args, **kwargs):
+        nonlocal cached_values_list
+
+        sublist = [*args]
+
+        if sublist in cached_values_list:
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+
+        cached_values_list += [[*args]]
+
+        return func(*args, **kwargs)
+
+    return inner
```

Comment:
your approach is incorrect: you should store arguments and the result of the function with these arguments, and return this result without calling function when arguments are the same.
hint: use dictionary as storage